main/ble_utils.c

```diff
@@ -873,22 +873,29 @@ ble_characteristic_t *ble_device_characteristic_add(ble_service_t *service,
     characteristic->handle = handle;
     characteristic->properties = properties;
     characteristic->client_config_handle = 0;
+    characteristic->index = 0;
+
+    for (cur = &service->characteristics; *cur; cur = &(*cur)->next){
+        if (ble_uuid_equal((*cur)->uuid, uuid)){
+            characteristic->index++;
+        }
+    };
 
-    for (cur = &service->characteristics; *cur; cur = &(*cur)->next);
     *cur = characteristic;
 
     return characteristic;
 }
 
 ble_characteristic_t *ble_device_characteristic_find_by_uuid(
-    ble_service_t *service, ble_uuid_t uuid)
+    ble_service_t *service, ble_uuid_t uuid, uint8_t index)
 {
     ble_characteristic_t *cur;
 
     for (cur = service->characteristics; cur; cur = cur->next)
     {
-        if (ble_uuid_equal(cur->uuid, uuid))
+        if (ble_uuid_equal(cur->uuid, uuid) && cur->index == index){
             break;
+        }
```

Comment:
No need for the curly braces here

Reply:
There may be no need, but it is better to put them regardless since it cleaner and less error-prone...

Reply:
My conventions are to use curly braces only if the condition or inner block are more than one line. Please make sure the new code follows those conventions.